src/data/monoid_algebra.lean

```diff
@@ -660,7 +661,7 @@ def lift [comm_semiring k] [add_monoid G] {R : Type u₃} [semiring R] [algebra
   left_inv := λ f, begin ext x, simp [sum_single_index] end,
   right_inv := λ F,
     begin
-      ext f,
+      apply alg_hom.ext, intro f,
```

Comment:
There seems to be a bug in `ext` here. If I leave this line as `ext f`, then it runs forever at this line, apparently because of the irrelevant `ext` lemma (irrelevant because it's about `monoid_algebra`, not `add_monoid_algebra`) which I've added above.

@cipher1024, do you know what's happening here?

Reply:
I've never seen that behaviour specifically but I've been meaning to improve `ext` for a while. I'll have a look and get back to you.